testo.d/lowranceusr.test

```diff
@@ -1,36 +1,150 @@
-bincompare ${TMPDIR}/enchilada1.usr ${REFERENCE}/ignoreicons.usr
+gpsbabel -i lowranceusr,ignoreicons -f ${REFERENCE}/lowrance-all.usr -o gpx,elevprec=6 -F ${TMPDIR}/lowrance-enchilada.gpx
+gpsbabel -i gpx -f ${TMPDIR}/lowrance-enchilada.gpx -o lowranceusr,wversion=2 -F ${TMPDIR}/lowrance-enchilada1.usr
+bincompare ${TMPDIR}/lowrance-enchilada1.usr ${REFERENCE}/lowrance-ignoreicons.usr
+
 #
-# Another variation of Lowrance.  Compare v2 and v3.  These original
+# Another variation of Lowrance.  Compare v2 and v3.  These reference
 # files were saved on the same units as v2 and v3.
 #
-${PNAME} -i lowranceusr -f ${REFERENCE}/lowrance-v2.usr -o unicsv,utc=0 -F ${TMPDIR}/lowrance-v2-unicsv.txt
-${PNAME} -i lowranceusr -f ${REFERENCE}/lowrance-v3.usr -o unicsv,utc=0 -F ${TMPDIR}/lowrance-v3-unicsv.txt
+gpsbabel -i lowranceusr -f ${REFERENCE}/lowrance-v2.usr -o unicsv,utc=0 -F ${TMPDIR}/lowrance-v2-unicsv.txt
+gpsbabel -i lowranceusr -f ${REFERENCE}/lowrance-v3.usr -o unicsv,utc=0 -F ${TMPDIR}/lowrance-v3-unicsv.txt
+# Compare generated to reference
 compare ${REFERENCE}/lowrance-v2-unicsv.txt ${TMPDIR}/lowrance-v2-unicsv.txt
+# Compare generated to reference
 compare ${REFERENCE}/lowrance-v3-unicsv.txt ${TMPDIR}/lowrance-v3-unicsv.txt
+# cut the depth column from V3 and then compare the resultant v3 with v2
+tail --lines=+2 ${REFERENCE}/lowrance-v2-unicsv.txt | sed "s/[0-9]*,//" | sort --field-separator=, --key=3 > ${TMPDIR}/lowrance-v2-unicsv-sorted.txt
+tail --lines=+2 ${REFERENCE}/lowrance-v3-unicsv.txt | sed "s/[0-9]*,//" | sort --field-separator=, --key=3 | cut --delimiter=, --complement -f 6 > ${TMPDIR}/lowrance-v3-unicsv-nodepth.txt
+compare ${TMPDIR}/lowrance-v2-unicsv-sorted.txt ${TMPDIR}/lowrance-v3-unicsv-nodepth.txt
+
 # Ideally, there'd be a test for v2 vs. v3 writes, but their numeric
 # instability makes this icky.
 
+#
+# ${REFERENCE}/lowrance-hook2-v2.usr, ${REFERENCE}/lowrance-hook2-v3.usr,
+# ${REFERENCE}/lowrance-hook2-v4.usr, ${REFERENCE}/lowrance-hook2-v5.usr, ${REFERENCE}/lowrance-hook2-v6.usr 
+# AND ${REFERENCE}/lowrance-hook2.gpx were generated on a single Lowrance Hook2 System.
+# This system has the ability to select any one of those formats for database export.
+# All files should contain the same basic data with increasing amounts of detail as you
+# move from USR v2 => USR v3 => USR v4 => USR v5 => USR v6
+#
+# Perform USR version 4 test
+#         -------------
+#
+gpsbabel -i lowranceusr -f ${REFERENCE}/lowrance-v4.usr -o gpx -F ${TMPDIR}/lowrance-v4.gpx
+compare ${REFERENCE}/lowrance-v4.gpx ${TMPDIR}/lowrance-v4.gpx
+
+#
+# Perform USR version 5 test
+#         -------------
+#
+gpsbabel -i lowranceusr -f ${REFERENCE}/lowrance-v5.usr -o gpx -F ${TMPDIR}/lowrance-v5.gpx
+compare ${REFERENCE}/lowrance-v5.gpx ${TMPDIR}/lowrance-v5.gpx
+
+# Should have generated the same GPX file as USR version 4 but the order is different for some reason
+# so use another way to compare to ensure have EXACTLY the same data
+#
```

Comment:
Interesting.  Maybe we shouldn't be storing rte_head->rte_num values at all if the order we retrieve the routes/trails doesn't mean anything.

If didn't store rte_num values then I have enhancements to the sort filter that could resort the routes & tracks by name, description or number.  The sort filter can already sort the waypoints by time, etc.

Reply:
Being able to sort would definitely simplify the verification tasks, and I could see where there would be advantages when moving from one vendor to another.  Right now there is so much loss of Waypoint, Trail and Route information moving from USR to GPX, or even USRx to USRy, that the change would seem to only benefit the validation process.  I'd definitely support adding the option but without significant support from Lowrance on decoding the USR formats it wouldn't be something that many folks would use at least for sorting data (which is not an option they have on their units).

Reply:
I merged #273.  I think it is of general usefulness, and if you want to use it for verification of lowranceusr it's available.  However, what you have works fine, and there is not need to change it.
For example
gpsbabel -i lowrancusr -f ${REFRENCE}/lowrance-v5.usr -x sort,rtename -o gpx -F lowrance-v5.gpx

However, because the route number (rte_num) values reflect the order in the usr file, while the route names don't, the gpx files still won't match between usr versions.